File: InventoryApp.Data/Models/InventoryDeliveryVoucher.cs
using InventoryApp.Data.Helper;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InventoryApp.Data.Models
{
    [Table("InventoryDeliveryVoucher")]
    public class InventoryDeliveryVoucher : EntityBase
    {
        [Key]
        public Guid Id { get; set; }
        public string Code { get; set; }
        public Guid WarehouseId { get; set; }
        public Warehouses? Warehouse { get; set; }
        public Guid BranchId { get; set; }
        public Branches? Branch { get; set; }
        public DateTime? GoodsIssueDate { get; set; }
        public int? OrderId { get; set; }
        public Order Order { get; set; }
        public Guid UserDeliveryId { get; set; }
        public Users? UserDelivery { get; set; }
        public List<InventoryDeliveryVoucherDetail> Details { get; set; }  
    }
}

File: InventoryApp.Client/src/app/module/inventory-delivery-voucher/model/inventory-delivery-voucher.ts
import { Branch } from "../../branch/model/branch";
import { Customer } from "../../business-partner/customer/model/customer";
import { CustomerOrder } from "../../customer-order/model/customer-order";
import { Order } from "../../order/model/order";
import { User } from "../../user/model/user";
import { Warehouse } from "../../warehouse/model/warehouse";
import { InventoryDeliveryVoucherDetail } from "./inventory-delivery-voucher-detail";

export interface InventoryDeliveryVoucher{
    id:string;
    code:string;
    orderId:string;
    order:CustomerOrder;
    branchId:string;
    branch:Branch;
    warehouseId:string;
    warehouse:Warehouse;
    userDeliveryId:string;
    userDelivery:User;
    goodsIssueDate:string;
    customer: Customer;
    details: InventoryDeliveryVoucherDetail[];
}